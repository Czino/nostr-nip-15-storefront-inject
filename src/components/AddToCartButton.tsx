export const AddToCartButton = ({ enabled = false }) => (
    <button
        disabled={!enabled}
        className={['w-full bg-highlight-1 p-3 rounded text-xl text-light-1', !enabled ? 'opacity-70' : ''].join(' ')}
    >
        {enabled ? 'Get in touch' : 'Out of stock'}
    </button>
)
