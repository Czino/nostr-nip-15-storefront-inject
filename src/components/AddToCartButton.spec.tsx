import { render } from '@testing-library/react'
import { AddToCartButton } from './AddToCartButton'

describe('AddToCartButton', () => {
    it('renders correctly when disabled', () => {
        const { asFragment } = render(<AddToCartButton enabled={false} />)
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders correctly when enabled', () => {
        const { asFragment } = render(<AddToCartButton enabled={true} />)
        expect(asFragment()).toMatchSnapshot()
    })
})
