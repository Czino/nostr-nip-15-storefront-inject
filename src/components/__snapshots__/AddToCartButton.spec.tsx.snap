// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton renders correctly when disabled 1`] = `
<DocumentFragment>
  <button
    class="w-full bg-highlight-1 p-3 rounded text-xl text-light-1 opacity-70"
    disabled=""
  >
    Out of stock
  </button>
</DocumentFragment>
`;

exports[`AddToCartButton renders correctly when enabled 1`] = `
<DocumentFragment>
  <button
    class="w-full bg-highlight-1 p-3 rounded text-xl text-light-1 "
  >
    Get in touch
  </button>
</DocumentFragment>
`;
